torch/csrc/jit/backends/backend.cpp

```diff
@@ -0,0 +1,37 @@
+#include <torch/csrc/jit/backends/backend.h>
+
+namespace torch {
+namespace jit {
+namespace {
+using BackendList = std::vector<Backend>;
+
+// Registry class for backends.
+struct BackendRegistry {
```

Comment:
I'm reading https://github.com/pytorch/pytorch/pull/34140 and wonder if we can just use the custom class registry mechanism there? @jamesr66a

Reply:
@yinghai yes, @suo and I have gone over the design and made changes to harmonize this with the custom classes API. Those adjustments will likely occur after this PR is landed